server/src/services/album.service.ts

```diff
@@ -146,8 +146,8 @@ export class AlbumService {
     const album = await this.findOrFail(id, { withAssets: true });
 
     if (dto.albumThumbnailAssetId) {
-      const valid = await this.albumRepository.hasAsset({ albumId: id, assetId: dto.albumThumbnailAssetId });
-      if (!valid) {
+      const results = await this.albumRepository.getAssetIds(id, [dto.albumThumbnailAssetId]);
```

Comment:
`hasAsset` excluded deleted entities, while `getAssetIds` includes them. Will this still behave as expected?

Reply:
I guess that would technically let you set the thumbnail to an id that is in the album but deleted?

Reply:
While that might not be ideal, web and mobile hide deleted assets in albums so you won't end up in this situation. The invalid thumbnail check for albums also doesn't exclude deleted assets, so it should be fine.

Reply:
That is a whole separate class of problems IMO. We might want to re-asses how deleted images are handled in albums, etc.

Reply:
Yeah I agree, that's beyond the scope of this PR